tests/watermarking/test_watermarking.py

```
@@ -0,0 +1,143 @@
+    monkeypatch.setattr(Sampler, "add_request", lambda *args, **kwargs: None)
+    monkeypatch.setattr(
+        "vllm.v1.watermarking.gpu_sampler.logger.warning_once", messages.append
+    )
+
+    sampler.add_request(0, 1, SamplingParams(temperature=0))
+    sampler.add_request(0, 1, SamplingParams(temperature=1))
+    sampler.add_request(0, 1, SamplingParams(temperature=0, watermarking=False))
+
+    assert messages == [
+        "Watermarking is enabled, but greedy decoding (temperature=0) cannot be "
+        "watermarked. This request will use ordinary greedy sampling."
```

Ruff (0.16.3)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)